src/lib/ext-monitor.js

```diff
@@ -81,6 +89,8 @@ export default class ExtensionMonitor {
     sendAllLogs: () => ({ existingLogs: this.logs }),
     loadLogs: (requestParams) => this.loadLogs(requestParams),
     getLoadedLogs: (requestParams) => this.getLoadedLogs(requestParams),
+    devToolsPanelShown: () => (this.isDevToolPanelOpen = true),
+    devToolsPanelHidden: () => (this.isDevToolPanelOpen = false),
```

Comment:
This part seems to not be taking into account that we may have multiple extension devtools panels opened at the same time (as I described in this comment https://github.com/mozilla/extension-activity-monitor/issues/22#issuecomment-667996573), and so when one of the extensions devtools panels is hidden this is going to set `this.isDevToolsPanelOpen` to `false` even if another devtools panel has not been hidden.

would you mind to describe me what is the issue and how you are trying to solve it at the moment?

Reply:
These were for keeping track of devtools panel. If there was any active devtools panel, we would be sending real-time logs there. 
Currently, I have applied a different approach to fix this, considering we have multiple extension devtools panels opened at the same time. I am storing the tabIds where activityLog devtools panel is shown and remvoing tabIds where activityLog devtools panel is hidden.

Reply:
@atiqueahmedziad I'm not sure that the devtools panel's `onShown` and `onHidden` events are the right way of keeping track if the devtools panel does still exist (hidden doesn't mean that we have destroyed it).

Another approach to keep track of that would be to make the activitylog page to connect a port if it does want the background page to send real time messages, that port would be automatically disconnected if the extension page or extension panel has been destroyed and so we could use that to detect when we don't have any page listening for realtime logs anymore (we don't need to send the real time logs over the `runtime.Port`, we can keep broadcasting them using `browser.runtime.sendMessage` as we are doing at the moment, and we could be using the ports just to keep track of the extension contexts that are still alive)

The flow would likely be something like the following:

- the activitylog page detects that it does want to receive the realtime logs and so it calls `browser.runtime.connect({name: "monitor-realtime-logs"})`
- on the background page side, ext-monitor would be registering a listener on the `browser.runtime.onConnect` API event, and if the received port does have the expected name (`"monitor-realtime-logs"` in the example above) and if it does:
  - it keeps track of the port (e.g. in a `Set`)
  - it subscribe a listener on `port.onDisconnect`, and remove the port from the data structure where we keep track of them (e.g. from the `Set` I did mention before)
- ext-monitor will then be broadcasting the logs collected in real time if there are `runtime.Port` instances still alive (still part of the `Set` I did mention above)

How that sounds to you?

Some references to the APIs I did mention:
- https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/runtime/connect
- https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/runtime/onConnect
- https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/runtime/Port

Reply:
Thanks @rpl for explaining in detail, I have implemented it :-)